src/pages/Data/BuyDataWithAirtime.tsx

```diff
@@ -43,47 +52,103 @@ export const BuyDataWithAirtime: React.FC = () => {
     },
   } = useGlobalStore();
 
+  const { data: dataEligibility } = useFetch<BorrowEligibilityResp>(
+    'Mobility.Account/api/data/GetBorrowingEligibility',
+  );
+
+  const { data: bundlesData } = useFetch<bundlesResp>(
+    'Mobility.Account/api/Data/GetBundle',
+  );
+
+  const [dataPlans, setDataPlans] = useState<
+    {
+      label: string;
+      value: string;
+    }[]
+  >([]);
+
+  useEffect(() => {
+    if (bundlesData) {
+      const plansResults = bundlesData?.result?.map((option) => ({
+        label: option.description,
+        value: String(option.dataPlanId),
+      }));
+
+      setDataPlans(plansResults);
+    }
+  }, [bundlesData]);
+
+  const [mobileNumbers, setMobileNumbers] = useState<
+    {
+      label: string;
+      value: string;
+    }[]
+  >([]);
+
+  useEffect(() => {
+    if (dataEligibility) {
+      const mobileResults = dataEligibility.result.borrowingOptions.map(
+        (option) => ({
+          label: option.mobileNumber,
+          value: option.mobileNumber,
+        }),
+      );
+
+      setMobileNumbers(mobileResults);
+    }
+  }, [dataEligibility]);
+
   const formik = useFormik({
     initialValues: {
-      recipientMobileNumber: '',
-      amount: '',
-      dataValue: '',
-      dataBundle: '',
+      mobileNumber: '',
+      beneficiaryMobileNumber: '',
+      bundleId: '',
     },
     validationSchema: Yup.object({
-      recipientMobileNumber: Yup.string()
+      mobileNumber: Yup.string()
         .matches(/^\d{11}$/, 'Must be an 11 digit phone number')
         .required('This field is required'),
-      voucherPin: Yup.string()
-        .min(16, 'Must be a 16 digit number')
-        .max(16, 'Must be a 16 digit number')
+      beneficiaryMobileNumber: Yup.string()
+        .matches(/^\d{11}$/, 'Must be an 11 digit phone number')
         .required('This field is required'),
+      bundleId: Yup.string().required('This field is required'),
     }),
     onSubmit: async (formData) => {
       setShowConfirmationModal(true);
     },
   });
 
   const handleTabChange = () => {
-    formik.resetForm();
     if (activeTab === 1) {
+      formik.setFieldValue('beneficiaryMobileNumber', '');
       return setactiveTab(2);
     }
+    formik.setFieldValue('beneficiaryMobileNumber', formik.values.mobileNumber);
     return setactiveTab(1);
   };
 
   const [showConfirmationModal, setShowConfirmationModal] = useState(false);
   const [showSuccessModal, setShowSuccessModal] = useState(false);
 
-  const handleRechargeWithPin = async () => {
-    const response = await rechargeWithPin(formik.values);
+  const handleBuyDataWithAirtime = async () => {
+    const response = await buyDataWithAirtime(formik.values);
```

Comment:
You will need to wrap this in a try catch for now. usePost throws, so we may have errors here if this fails

Reply:
I've updated as requested